projects/dotcms-field-elements/src/components/dot-binary-file/dot-binary-file.tsx

```diff
@@ -0,0 +1,312 @@
+                } else {
+                    this.clearField(this.validationMessage);
+                }
+            }
+        });
+    }
+
+    // only supported in iOS.
+    private pasteHandler(): void {
+        this.el.addEventListener('paste', (evt: ClipboardEvent) => {
+            const clipboardData: DataTransfer = evt.clipboardData;
+            if (clipboardData.items.length) {
+                if (this.isPastingFile(clipboardData)) {
+                    this.handleFilePaste(clipboardData.items);
+                } else {
+                    this.handleURLPaste(clipboardData.items[0]);
+                }
+            }
+        });
+    }
+
+    private changeHandler(event: Event): void {
+        const input: HTMLInputElement = event.srcElement as HTMLInputElement;
+        if (this.isFileAllowed(input.files[0].name)) {
+            this.setValue(input.files[0]);
+            this.setFileDisplayName(input.files[0].name);
+        } else {
+            event.preventDefault();
+            this.clearField(this.validationMessage);
+        }
+    }
+
+    private handleFilePaste(items: DataTransferItemList) {
+        const clipBoardFile = items[1].getAsFile();
+        items[0].getAsString((fileName: string) => {
+            if (this.isFileAllowed(fileName)) {
+                this.setValue(clipBoardFile);
+            } else {
+                this.clearField(this.validationMessage);
+            }
+        });
+    }
+
+    private handleURLPaste(item: DataTransferItem) {
+        item.getAsString((fileURL: string) => {
+            if (this.isValidURL(fileURL)) {
+                this.setValue(fileURL);
+            } else {
+                this.clearField(this.URLValidationMessage);
+            }
+        });
+    }
+
+    private clearField(validationMessage: string) {
+        this.setValue(null, validationMessage);
+        this.setFileDisplayName('');
+    }
+
+    private isValidURL(url: string): boolean {
+        return URL_REGEX.test(url);
+    }
+
+    private isPastingFile(data: DataTransfer): boolean {
+        return !!data.files.length;
+    }
+
+    private keyDownHandler(evt: KeyboardEvent): void {
+        if (evt.key === 'Backspace') {
+            this.required ? this.setValue(null, this.requiredMessage) : this.setValue(null);
+        }
+    }
+
+    private buttonHandler(): void {
+        const fileInput: HTMLInputElement = this.el.querySelector('dot-label input[type="file"]');
+        fileInput.click();
+    }
+
+    private setFileDisplayName(name: string): void {
+        const textInput: HTMLInputElement = this.el.querySelector('dot-label input[type="text"]');
+        textInput.value = name;
+    }
+
+    private isFileAllowed(name: string): boolean {
+        const extension = name.substring(name.indexOf('.'), name.length);
+        if (this.allowedFileTypes.length === 0 || this.allowedFileTypes.indexOf('*') === 0) {
+            return true;
```

Comment:
```suggestion
            return return this.allowedFileTypes.length === 0 || this.allowedFileTypes.indexOf('*') === 0 || this.allowedFileTypes.indexOf(extension) >= 0;
```

Still think the conditions should be extracted.

Reply:
but you have the else too, I think it have to be something like:

```suggestion
            return this.AllowAnyFile() || this.allowedFileTypes.indexOf(extension) >= 0;
```

and allowAnyFile:

```
allowAnyFile() :  boolean {
     return this.allowedFileTypes.length === 0 || this.allowedFileTypes.indexOf('*') === 0
}
```